test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -223,6 +226,133 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+	crdClient := data.e2eTestData.CrdClient
+	
+	return wait.PollUntilContextTimeout(context.Background(), 5*time.Second, 40*time.Second, true, func(ctx context.Context) (bool, error) {
+		for _, iface := range interfaces {
+			ipPoolName, exists := ipPools[iface]
+			if !exists {
+				return false, fmt.Errorf("no IPPool found for interface %s", iface)
+			}
+
+			podIP, exists := podIPsAndIfaces[iface]
+			if !exists {
+				return false, fmt.Errorf("no IP found for interface %s", iface)
+			}
+
+			ipPool, err := crdClient.CrdV1beta1().IPPools().Get(ctx, ipPoolName[0], metav1.GetOptions{})
+			if err != nil {
+				return false, fmt.Errorf("failed to get IPPool %s: %w", ipPoolName[0], err)
+			}
+
+			for _, ipAddress := range ipPool.Status.IPAddresses {
+				if podIP.String() == ipAddress.IPAddress {
+					return false, nil
+				}
+			}
+			logs.Infof("Released PodIP: %v, Interface: %s", podIP, iface)
+		}
+		return true, nil
+	})
+}
+
+func checkIPsAndIfaces(before, after map[string]net.IP) bool {
+	if len(before) != len(after) {
+		return false
+	}
+	for iface, beforeIP := range before {
+		afterIP, exists := after[iface]
+		if !exists || !beforeIP.Equal(afterIP) {
+			return false
+		}
+	}
+	return true
+}
+
+// reconcilationAfterAgentRestart verifies Interface cleanup and IP release.
+func (data *testData) reconcilationAfterAgentRestart(t *testing.T) error {
+	vlanPod := data.pods[1]
+	interfaces := []string{"eth1", "eth2"}
+
+	beforeAgentRestartIPsAndIfaces, err := data.listPodIPs(vlanPod)
+	if err != nil {
+		return fmt.Errorf("failed to get Pod IP before agent restart: %w", err)
+	}
+
+	// Restarting the Antrea agent
+	if err := data.e2eTestData.RestartAntreaAgentPods(30 * time.Second); err != nil {
+		t.Fatalf("Failed to restart Antrea agent Pods: %v", err)
+	}
+
+	afterAgentRestartIPsAndIfaces, err := data.listPodIPs(vlanPod)
+	if err != nil {
+		return fmt.Errorf("failed to get Pod IP after agent restart: %w", err)
+	}
+
+	if !checkIPsAndIfaces(beforeAgentRestartIPsAndIfaces, afterAgentRestartIPsAndIfaces) {
+		return fmt.Errorf("Pod: %s, Interfaces and IPs mismatch after agent restart", vlanPod.podName)
+	}
+
+	logs.Infof("Pod IPs and Interfaces before agent restart:")
+	for iface, ip := range beforeAgentRestartIPsAndIfaces {
+		logs.Infof("  Interface: %s, IP: %s", iface, ip)
+	}
+
+	logs.Infof("Pod IPs and Interfaces after agent restart:")
+	for iface, ip := range afterAgentRestartIPsAndIfaces {
+		logs.Infof("  Interface: %s, IP: %s", iface, ip)
+	}
+
+	ipPools, err := data.getIPPoolNames(interfaces, vlanPod, "default")
+	if err != nil {
+		return fmt.Errorf("failed to get IPPool: %w", err)
+	}
+
+	// Remove Pod and check IP released or not.
+	if err := data.e2eTestData.DeletePodAndWait(defaultTimeout, vlanPod.podName, data.e2eTestData.GetTestNamespace()); err != nil {
+		return fmt.Errorf("failed to delete Pod %s: %w", vlanPod.podName, err)
+	}
+
+	return data.checkIPReleased(ipPools, beforeAgentRestartIPsAndIfaces, interfaces)
```

Comment:
A reminder for checking OVS ports deleted.

Reply:
done